pkg/serverless/trace/propagation/extractor_test.go

```diff
@@ -407,6 +407,72 @@ func TestExtractorExtract(t *testing.T) {
 			expNoErr: true,
 		},
 
+		// events.EventBridgeEvent
+		{
+			name: "eventbridge-event-empty",
+			events: []interface{}{
+				events.EventBridgeEvent{},
+			},
+			expCtx:   nil,
+			expNoErr: false,
+		},
+		{
+			name: "eventbridge-event-with-dd-headers",
+			events: []interface{}{
+				events.EventBridgeEvent{
+					Detail: struct {
+						TraceContext map[string]string `json:"_datadog"`
+					}{
+						TraceContext: headersMapDD,
+					},
+				},
+			},
+			expCtx:   ddTraceContext,
+			expNoErr: true,
+		},
+		{
+			name: "eventbridge-event-with-all-headers",
+			events: []interface{}{
+				events.EventBridgeEvent{
+					Detail: struct {
+						TraceContext map[string]string `json:"_datadog"`
+					}{
+						TraceContext: headersMapAll,
+					},
+				},
+			},
+			expCtx:   ddTraceContext,
+			expNoErr: true,
+		},
+		{
+			name: "eventbridge-event-with-w3c-headers",
+			events: []interface{}{
+				events.EventBridgeEvent{
+					Detail: struct {
+						TraceContext map[string]string `json:"_datadog"`
+					}{
+						TraceContext: headersMapW3C,
+					},
+				},
+			},
+			expCtx:   w3cTraceContext,
+			expNoErr: true,
+		},
```

Comment:
Could we add a W3C test with resource name + start time? Not sure where it should live, but I want to make sure we are covering that behavior.

Reply:
Sure, I added a test in my [latest commit](https://github.com/DataDog/datadog-agent/pull/29414/commits/06c5ae4a0a28ccacba1336f15ea0c24bdca44e4e)